app/Services/TaskScheduler/TaskSchedulerService.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+
+namespace App\Services\TaskScheduler;
+
+
+use App\Export\CSV\ClientExport;
+use App\Export\CSV\ContactExport;
+use App\Export\CSV\CreditExport;
+use App\Export\CSV\DocumentExport;
+use App\Export\CSV\ExpenseExport;
+use App\Export\CSV\InvoiceExport;
+use App\Export\CSV\InvoiceItemExport;
+use App\Export\CSV\PaymentExport;
+use App\Export\CSV\ProductExport;
+use App\Export\CSV\QuoteExport;
+use App\Export\CSV\QuoteItemExport;
+use App\Export\CSV\RecurringInvoiceExport;
+use App\Export\CSV\TaskExport;
+use App\Http\Requests\Report\GenericReportRequest;
+use App\Http\Requests\Report\ProfitLossRequest;
+use App\Http\Requests\TaskScheduler\CreateScheduledTaskRequest;
+use App\Http\Requests\TaskScheduler\UpdateScheduledJobRequest;
+use App\Http\Requests\TaskScheduler\UpdateScheduleRequest;
+use App\Jobs\Report\ProfitAndLoss;
+use App\Models\ScheduledJob;
+use App\Models\Scheduler;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Model;
+use Symfony\Component\HttpFoundation\Request;
+
+class TaskSchedulerService
+{
+    public Scheduler $scheduler;
+
+    public function __construct(Scheduler $scheduler)
+    {
+        $this->scheduler = $scheduler;
+    }
+
+    public function store(Scheduler $scheduler, CreateScheduledTaskRequest $request)
+    {
+        $scheduler->paused = $request->get('paused', false);
+        $scheduler->archived = (bool)$request->get('archived', false);
+        $scheduler->start_from = $request->get('start_from') ? Carbon::parse((int)$request->get('start_from')) : Carbon::now();
+        $scheduler->repeat_every = $request->get('repeat_every');
+        $scheduler->scheduled_run = $request->get('start_from') ? Carbon::parse((int)$request->get('start_from')) : Carbon::now();;
+        $scheduler->save();
+
+        if ($this->createJob($request, $scheduler)) {
+            return response(['job_has_been_created'], 200);
+        }
+        return response(['failed_to_create_job'], 400);
+    }
+
+    public function update(Scheduler $scheduler, UpdateScheduleRequest $request)
+    {
+        $data = $request->validated();
+        if ($request->has('start_from')) {
+            $data['start_from'] = Carbon::parse((int)$request->get('start_from'));
```

Comment:
this should all be handled in the form request, use prepareForValidation() to set any calculated properties.